build_test/lib/src/in_memory_reader.dart

```diff
@@ -17,19 +17,17 @@ class InMemoryAssetReader implements AssetReader {
       : assets = sourceAssets ?? <AssetId, DatedValue>{};
 
   @override
-  Future<bool> hasInput(AssetId id) {
-    return new Future.value(assets.containsKey(id));
-  }
+  FutureOr<bool> canRead(AssetId id) => assets.containsKey(id);
```

Comment:
Discussed offline. hasInput is removed from AssetReader (very few people impacted but is technically breaking for client code, not just build system implementations) Only BuildStep keeps the old method.